### PR TITLE
Use db owner in java migration when needed

### LINKED ISSUE #4474: Run integrations tests as regular database user
### Problem

There have been a few regressions due to the integration tests running as the database owner.

### Solution

Change testcontainer configuration to create a new, non-owner database user and use it to run integration tests. Flyway migration tests should continue to run as database owner.

### Alternatives

_No response_

### LINKED ISSUE #4475: Run Java migrations as database owner
### Description

Flyway SQL migrations run as the database owner but in Java migrations we currently do not.

### Steps to reproduce

Create a Java migration that does an elevated operation like truncation

### Additional context

_No response_

### Hedera network

other

### Version

v0.64.1

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR improves the importer integration test to use different db users for migration and regular processing

- Create db owner `mirror_node` and importer user `mirror_importer` in postgres test container
- Create a separate `JdbcTemplate` bean with db owner credentials
- Use the db owner `JdbcTemplate` bean in java migrations when needed

**Related issue(s)**:

Fixes #4474
Fixes #4475

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/test/resources/db/migration/v1/V1.10.0.1__fix_regular_db_user_privileges.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+revoke all on all tables in schema public from ${db-user};
+alter default privileges in schema public revoke all on tables from ${db-user};
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/test/resources/db/migration/v1/V1.10.0.1__fix_regular_db_user_privileges.sql:2]: the migration only runs in importer integration test. it's used to revert the excessive privilege granted to `db-user` in migration V1.10.0

```
alter default privileges in schema public
    grant all on tables to ${db-user};
```

**Review by edwin-greene — DISMISSED** [review]
Looks good!

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/JdbcTemplateConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import com.hedera.mirror.importer.db.DBProperties;
+
+@Configuration
+public class JdbcTemplateConfiguration {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/JdbcTemplateConfiguration.java:33]: nit: With custom qualifier can make package private and final. Also make methods package private.
```suggestion
final class JdbcTemplateConfiguration {
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/BackfillTransactionHashMigration.java [hunk 3]
```diff
@@ -49,7 +52,8 @@ insert into transaction_hash (consensus_timestamp, hash)
     private final JdbcTemplate jdbcTemplate;
 
     @Lazy
-    public BackfillTransactionHashMigration(EntityProperties entityProperties, JdbcTemplate jdbcTemplate,
+    public BackfillTransactionHashMigration(EntityProperties entityProperties,
+                                            @Qualifier(JDBC_TEMPLATE_OWNER) JdbcTemplate jdbcTemplate,
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/BackfillTransactionHashMigration.java:56]: Would prefer a custom qualifier that can provide reuse in the future for non-JdbcTemplate use cases:

```java
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.PARAMETER})
public @interface Owner {
}
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/JdbcTemplateConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    public static final String JDBC_TEMPLATE_OWNER = "owner";
+
+    @Bean
+    @Primary
+    public JdbcTemplate jdbcTemplate(DataSource dataSource) {
+        return new JdbcTemplate(dataSource);
+    }
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/JdbcTemplateConfiguration.java:41]: Maybe we should duplicate the current `JdbcTemplateConfiguration` that we're replacing so it respects the existing properties. Can pull into a helper method.
```suggestion
	JdbcTemplate jdbcTemplate(DataSource dataSource, JdbcProperties properties) {
		JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
		JdbcProperties.Template template = properties.getTemplate();
		jdbcTemplate.setFetchSize(template.getFetchSize());
		jdbcTemplate.setMaxRows(template.getMaxRows());
		if (template.getQueryTimeout() != null) {
			jdbcTemplate.setQueryTimeout((int) template.getQueryTimeout().getSeconds());
		}
		return jdbcTemplate;
	}
```

File: hedera-mirror-importer/src/test/resources/db/scripts/init.sql [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+grant usage on schema public to public;
+revoke create on schema public from public;
+
+-- Grant readonly privileges
+grant connect on database mirror_node to readonly;
+grant select on all tables in schema public to readonly;
+grant select on all sequences in schema public to readonly;
+grant usage on schema public to readonly;
+alter default privileges in schema public grant select on tables to readonly;
+alter default privileges in schema public grant select on sequences to readonly;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/resources/db/scripts/init.sql:19]: nit: We can probably skip all the statements involving sequences considering we don't use sequences...

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/BackfillTransactionHashMigration.java (hunk 3 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/BackfillTransactionHashMigration.java]: added

File: hedera-mirror-importer/src/test/resources/db/scripts/init.sql (hunk 5 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/test/resources/db/scripts/init.sql]: removed those

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**